src/karapace/api/oidc/middleware.py

```diff
@@ -0,0 +1,59 @@
+"""
+Copyright (c) 2025 Aiven Ltd
+See LICENSE for details
+"""
+
+import jwt
+import logging
+from jwt import PyJWKClient, InvalidTokenError
+from starlette.responses import JSONResponse
+
+from src.karapace.core.auth import AuthenticationError
+from src.karapace.core.config import Config
+
+log = logging.getLogger(__name__)
+
+
+class OIDCMiddleware:
+    def __init__(self, app, config: Config):
+        self.app = app
+        self.config = config
+        self.jwks_url = config.sasl_oauthbearer_jwks_endpoint_url
+        self.issuer = config.sasl_oauthbearer_expected_issuer
+        self.audience = config.sasl_oauthbearer_expected_audience
+        self.claim_name = config.sasl_oauthbearer_sub_claim_name
+        self._jwks_client = PyJWKClient(self.jwks_url)
+
+    def validate_jwt(self, token: str) -> dict:
+        try:
+            signing_key = self._jwks_client.get_signing_key_from_jwt(token)
+            # Split the comma-separated audience string into a list and strip whitespace
+            if self.audience:
+                audiences = [aud.strip() for aud in self.audience.split(",") if aud.strip()]
```

Comment:
Is the ` if aud.strip()` needed? Why not splitting audience in the `__init__` method?